eogrow/pipelines/export_maps.py

```diff
@@ -76,54 +90,61 @@ def run_procedure(self) -> Tuple[List[str], List[str]]:
         if not successful:
             raise ValueError("Failed to extract tiff files from any of EOPatches.")
 
-        folder = self._get_output_folder()
+        feature_type, _ = self.config.feature
+        folder = self.storage.get_folder(self.config.output_folder_key)
         crs_eopatch_dict = self.eopatch_manager.split_by_utm(successful)
 
         # TODO: This could be parallelized per-crs
         for crs, eopatch_list in crs_eopatch_dict.items():
-            subfolder = f"UTM_{crs.epsg}"
+            output_folder = fs.path.join(folder, f"UTM_{crs.epsg}")
             # manually make subfolder, otherwise things fail on S3 in later steps
-            self.storage.filesystem.makedirs(fs.path.join(folder, subfolder), recreate=True)
+            self.storage.filesystem.makedirs(output_folder, recreate=True)
 
-            map_name = self.config.map_name or f"{self.config.feature[1]}.tiff"
-            merged_map_path = fs.path.join(folder, subfolder, map_name)
-            geotiff_paths = [fs.path.join(folder, self.get_geotiff_name(name)) for name in eopatch_list]
+            merged_map_path = fs.path.join(output_folder, self.get_geotiff_name("full_merged_map"))
+            exported_tiff_paths = [fs.path.join(folder, self.get_geotiff_name(name)) for name in eopatch_list]
 
-            temp_fs, geotiff_sys_paths, map_sys_path = self._prepare_files(geotiff_paths, merged_map_path)
+            filesystem, geotiff_paths, map_path = self._prepare_files(exported_tiff_paths, merged_map_path)
 
             merge_tiffs(
-                geotiff_sys_paths,
-                map_sys_path,
+                list(map(filesystem.getsyspath, geotiff_paths)),
+                filesystem.getsyspath(map_path),
                 overwrite=True,
                 nodata=self.config.no_data_value,
                 dtype=self.config.map_dtype,
             )
 
-            if self.config.cogify:
-                cogify_inplace(
-                    map_sys_path, blocksize=1024, nodata=self.config.no_data_value, dtype=self.config.map_dtype
-                )
+            output_paths: List[Tuple[str, Optional[dt.datetime]]]
+            if feature_type.is_timeless() or self.config.compress_temporally:
+                output_paths = [(map_path, None)]
+            else:
+                timestamp = self._load_timestamp(eopatch_list[0])  # we assume all eopatches share the same timestamp
+                output_paths = self._split_temporally(filesystem, map_path, timestamp, output_folder)
 
-            self._finalize_output_files(temp_fs, map_sys_path, merged_map_path)
+            if self.config.cogify:
+                resampling = "mode" if feature_type.is_discrete() else "bilinear"
```

Comment:
is this stable? can it happen that you have `[0.0, 1.0, 4.0, 10.0]` values which are actually discrete but still have a floating point? 

I don't know if it makes sense to solve for such cases anyway

Reply:
This uses `mode` for `FeatureType.MASK` and `FeatureType.MASK_TIMELESS`. Those two feature types are also checked to be discrete. The only other way of doing this would be to check the dtype at runtime i guess. Feels a bit hacky

Reply:
👍